### PR TITLE
feat(config): add a more explicative error to the validation

### PR DESCRIPTION
**Pull request by joshuachp**
The validation function now returns a ConfigValidationError, which contains more information about the error. This is useful for the user to understand what's wrong with the configuration file.

### CODE REVIEW

**Comment by joshuachp** [comment]
The error before:

```
Error: FatalError("Toml file does not contain valid options.")
```

and now:

```
Error: FatalError("Invalid configuration file: either the pairing token or credential secret must be provided")
```

File: src/main.rs [hunk 1]
```diff
@@ -93,7 +93,7 @@ async fn initialize_astarte_device_sdk(
     }
 
     if let Some(int_dir) = &msg_hub_opts.interfaces_directory {
-        device_sdk_opts = device_sdk_opts.interface_directory(int_dir)?;
+        device_sdk_opts = device_sdk_opts.interface_directory(&int_dir.to_string_lossy())?;
```
Comment (sorru94) [inline src/main.rs:96]: What is the purpose of this change?

**Review by harlem88 — APPROVED** [review]
